tests/filesystem.go

```diff
@@ -127,42 +127,60 @@ func pickPartition(t *testing.T, device string, partitions []*types.Partition, l
 	return ""
 }
 
-// createVolume will create the image file of the specified size, create a
-// partition table in it, and generate mount paths for every partition
-func createVolume(t *testing.T, ctx context.Context, tmpDirectory string, index int, imageFile string, size int64, partitions []*types.Partition) (err error) {
+// setupDisk creates a backing file then loop mounts it. It sets up the partitions and filesystems on that loop device.
+// It returns any error it encounters, but cleans up after itself if it errors out.
+func setupDisk(t *testing.T, ctx context.Context, disk *types.Disk, diskIndex int, imageSize int64, tmpDirectory string) (err error) {
 	// attempt to create the file, will leave already existing files alone.
 	// os.Truncate requires the file to already exist
-	var out *os.File
-	out, err = os.Create(imageFile)
-	if err != nil {
+	var (
+		out *os.File
+		tmp []byte
+	)
+	if out, err = os.Create(disk.ImageFile); err != nil {
 		return err
 	}
 	defer func() {
 		// Delete the image file if this function exits with an error
 		if err != nil {
-			os.Remove(imageFile)
+			os.Remove(disk.ImageFile)
 		}
 	}()
 	out.Close()
 
 	// Truncate the file to the given size
-	err = os.Truncate(imageFile, size)
-	if err != nil {
+	if err = os.Truncate(disk.ImageFile, imageSize); err != nil {
 		return err
 	}
 
-	err = createPartitionTable(t, ctx, imageFile, partitions)
+	// Attach the file to a loopback device
+	tmp, err = run(t, ctx, "losetup", "-Pf", "--show", "-b", "512", disk.ImageFile)
```

Comment:
Should sector size be added to `disk` and referenced here rather than the hardcoded 512?

Reply:
oh whoops the `-b` was meant to be left off; I added this while debugging something. I probably going to add another attribute to `tests/types.Disk` to specify block size later. I'll remove this.

Reply:
fixed